lib/process-release.js

```diff
@@ -0,0 +1,87 @@
+
+  // parse the version to normalize and ensure it's valid
+  var versionSemver = semver.parse(versionStr)
+  if (!versionSemver) {
+    return { versionStr: versionStr }
+  }
+  // flatten version into String
+  versionStr = versionSemver.version
+
+  var overrideDistUrl = gyp.opts['dist-url'] || gyp.opts.disturl
+  var iojs = !overrideDistUrl && !defaultRelease && semver.gte(versionStr, '1.0.0')
+  var name = (defaultRelease && defaultRelease.name) || (iojs ? 'iojs' : 'node')
+  var defaultDirUrl = (overrideDistUrl || iojs ? 'https://iojs.org/download/release' : 'https://nodejs.org/dist') + '/v' + versionStr + '/'
+  var baseUrl
+  var libUrl32
+  var libUrl64
+
+  if (defaultRelease && defaultRelease.headersUrl && !overrideDistUrl) {
+    baseUrl = url.resolve(defaultRelease.headersUrl, './')
+    libUrl32 = resolveLibUrl(name, defaultRelease.libUrl || baseUrl || defaultDirUrl, 32, versionStr)
+    libUrl64 = resolveLibUrl(name, defaultRelease.libUrl || baseUrl || defaultDirUrl, 64, versionStr)
+
+    return {
+      versionStr: versionStr,
+      versionSemver: versionSemver,
+      name: defaultRelease.name,
+      baseUrl: baseUrl,
+      tarballUrl: defaultRelease.headersUrl,
+      shasumsFile: 'SHASUMS256.txt',
+      shasumsUrl: url.resolve(baseUrl, 'SHASUMS256.txt'),
+      checksumAlgo: 'sha256',
+      versionDir: (defaultRelease.name != 'node' ? defaultRelease.name + '-' : '') + versionStr,
+      libUrl32: libUrl32,
+      libUrl64: libUrl64,
+      libPath32: path.relative(url.parse(baseUrl).path, url.parse(libUrl32).path),
+      libPath64: path.relative(url.parse(baseUrl).path, url.parse(libUrl64).path)
+    }
+  }
+
+  // distributions starting with 0.10.0 contain sha256 checksums
+  var checksumAlgo = semver.gte(versionStr, '0.10.0') ? 'sha256' : 'sha1'
+  var shasumsFile = (checksumAlgo === 'sha256') ? 'SHASUMS256.txt' : 'SHASUMS.txt'
+
+  baseUrl = defaultDirUrl
+  libUrl32 = resolveLibUrl(name, baseUrl, 32, versionStr)
+  libUrl64 = resolveLibUrl(name, baseUrl, 64, versionStr)
+
+  return {
+    versionStr: versionStr,
+    versionSemver: versionSemver,
+    name: name,
+    baseUrl: baseUrl,
+    tarballUrl: baseUrl + name + '-v' + versionStr + '.tar.gz',
+    shasumsFile: shasumsFile,
+    shasumsUrl: baseUrl + shasumsFile,
+    checksumAlgo: checksumAlgo,
+    versionDir: (name != 'node' ? name + '-' : '') + versionStr,
+    libUrl32: libUrl32,
+    libUrl64: libUrl64,
+    libPath32: path.relative(url.parse(baseUrl).path, url.parse(libUrl32).path),
+    libPath64: path.relative(url.parse(baseUrl).path, url.parse(libUrl64).path)
+  }
+}
+
+function resolveLibUrl (name, defaultUrl, bits, version) {
+  var base = url.resolve(defaultUrl, './')
+
+  if (!bitsre.test(defaultUrl)) {
+    // let's assume it's a baseUrl then
+    if (semver.gte(version, '1.0.0'))
+      return url.resolve(base, 'win-x' + bits +'/' + name + '.lib')
+    return url.resolve(base, (bits == 64 ? 'x64/' : '') + name + '.lib')
```

Comment:
@rvagg I had forgotten. I want to make what i think is a better suggestion here.

Can we just modify those old folders to give them a new directory structure? I don't think that should effect the SHASUMS, and would probably be more robust.